### PR TITLE
git-chain: init at 0-unstable-2024-08-09

### PR DESCRIPTION
**Pull request by bcyran**
## Description of changes

Adding [`git-chain`](https://github.com/dashed/git-chain) - a tool for working with a chain of dependent git branches. Simplifies some tedious tasks like rebasing many branches after pulling changes from the main branch.

FYI @dashed 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gi/git-chain/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  git,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "git-chain";
+  version = "0-unstable-2024-09-09";
+
+  src = fetchFromGitHub {
+    owner = "dashed";
+    repo = pname;
```
[the remark this answers is not in the document]
Comment (bcyran) [reply pkgs/by-name/gi/git-chain/package.nix]: Done!

File: pkgs/by-name/gi/git-chain/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  cargoHash = "sha256-f254nxULp+8IDvLJU23YJGSlvMKaJm5ZU/7WVZztuR0=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
+
+  nativeCheckInputs = [ git ];
+
+  meta = with lib; {
+    description = "Tool for rebasing a chain of local git branches";
```
[the remark this answers is not in the document]
Comment (bcyran) [reply pkgs/by-name/gi/git-chain/package.nix:30]: This project doesn't provide any changelog or any kind of releases, not even tags. Not sure what I could link as a changelog.

**Comment by bcyran** [comment]
@Bot-wxt1221 Do I need to do anything after addressing the comments, like requesting review? And what's up with ofborg's internal error?

Not trying to rush you or anything, I just don't know the usual workflow here.

**Comment by bcyran** [comment]
@Bot-wxt1221 Thanks for help and the review!

Added an update script. Testing it actually revealed that I initially made a mistake in the version so it was a good call. Would appreciate you taking a final look.

**Review by Aleksanaa — APPROVED** [review]

**Comment by bcyran** [comment]
@Bot-wxt1221 @Aleksanaa Hi guys! Sorry to bother you but is it normal for PRs to get stuck for so long? Is there anything I can do to push this forward? Anything I can do to avoid this in the future?

**Comment by uncenter** [comment]
> is it normal for PRs to get stuck for so long? Is there anything I can do to push this forward?

Sometimes unfortunately. With such a high volume of PRs, some get missed. It helps to be really active in promoting/getting reviewers to approve and bump your PR in the list. As you can see by https://github.com/NixOS/nixpkgs/pull/342850#issuecomment-2366247982, there are some forum threads and also Discord channels to request reviews.

File: pkgs/by-name/gi/git-chain/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  git,
+  nix-update-script,
+}:
```
Comment (uncenter) [inline pkgs/by-name/gi/git-chain/package.nix:9]:
```suggestion
  stdenv,
  darwin,
}:
```

File: pkgs/by-name/gi/git-chain/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    repo = "git-chain";
+    rev = "4fee033ea1ee51bbb6b7f75411f0f4f799aea1e2";
+    hash = "sha256-wQZXixg7mCBUo18z/WCkTWW3R0j2jxs8t1yaQzY3Eu4=";
+  };
+
+  cargoHash = "sha256-pRxOrlDgfSpUBY2WKfoIH9ngLzb2noiLqxA3/6s+mRw=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
```
Comment (uncenter) [inline pkgs/by-name/gi/git-chain/package.nix:26]:
```suggestion
  buildInputs =
    [ openssl ]
    ++ lib.optionals stdenv.hostPlatform.isDarwin (
      with darwin.apple_sdk.frameworks;
      [
        Security
      ]
    );
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Comment by bcyran** [comment]
Thanks for help @uncenter!

I applied and pushed the requested changes around 12 hours ago but for some reason GitHub is stuck at "Processing updates". I tried rebasing and pushing a few more times but to no avail. I also just noticed that apparently there are still 2 pending CI jobs queued for 20 days now? As a person used to GitLab I'm quite confused by all of this. Do you have any advice how to proceed?

**Comment by uncenter** [comment]
Hmm are you getting an error locally when trying to push? Can you explain more about your issue? Is it still occuring?

**Comment by bcyran** [comment]
@uncenter The problem was that the pushed changes were visible on my branch but not in the PR. I rebased and pushed one more time right now and the problem is gone so I guess it was some kind of GH lag.

Anyway, your suggestions are applied!

**Pull request merged**